src/metabase/api/common.clj

```diff
@@ -26,6 +26,10 @@
    ex. `@*current-user*`"
   (atom nil)) ; default binding is just something that will return nil when dereferenced
 
+(def ^:dynamic ^Boolean *is-superuser?*
+  "Is the current user a superuser?"
+  false)
+
```

Comment:
Yup the API call middleware binds this to `true`/`false` as appropriate for the current user